### PR TITLE
Prevent defining illegal registry key

### PR DESCRIPTION
**Pull request by arunans23**
Prevent defining illegal registry key

### CODE REVIEW

File: components/mediation/registry/org.wso2.micro.integrator.registry/src/main/java/org/wso2/micro/integrator/registry/MicroIntegratorRegistry.java [hunk 1]
```diff
@@ -861,14 +862,22 @@ private String resolveRegistryURI(String key) {
...
-
             resolvedPath = registryRoot + resourcePath;
-        }
 
+            //Test whether registry key has any illegel access
+            File resolvedPathFile = new File(resolvedPath);
+            File registryRootFile = new File(registryRoot);
+            try {
+                if (!resolvedPathFile.getCanonicalPath().startsWith(registryRootFile.getCanonicalPath())) {
+                    handleException("The registry key defined is illegal");
```
Comment (sdkottegoda) [inline components/mediation/registry/org.wso2.micro.integrator.registry/src/main/java/org/wso2/micro/integrator/registry/MicroIntegratorRegistry.java:872]: Should we log the resolved path? and the registry root path?

Comment (arunans23) [reply components/mediation/registry/org.wso2.micro.integrator.registry/src/main/java/org/wso2/micro/integrator/registry/MicroIntegratorRegistry.java:872]: In this case, we do not need to. This method will be called several times for various resources. Maybe we can add a debug log. WDYT?

Comment (Nirothipan) [reply components/mediation/registry/org.wso2.micro.integrator.registry/src/main/java/org/wso2/micro/integrator/registry/MicroIntegratorRegistry.java:872]: +1 to print registry root path but I'm not sure we can print the illegal path in the logs.

Comment (arunans23) [reply components/mediation/registry/org.wso2.micro.integrator.registry/src/main/java/org/wso2/micro/integrator/registry/MicroIntegratorRegistry.java]: Logged the registry key

**Pull request merged**